tests/integration/test_images.py

```diff
@@ -17,128 +20,268 @@ def intersect_tasks(
-            # No mpas_analysis
-            available_tasks = ["e3sm_diags", "global_time_series", "ilamb"]
-            tasks_to_run = intersect_tasks(available_tasks, requested_tasks)
-            set_up_and_run_image_checker(
+        )
+
+    if "weekly_legacy_3.0.0_bundles" in expansions["cfgs_to_run"]:
+        # No mpas_analysis
+        available_tasks = ["e3sm_diags", "global_time_series", "ilamb"]
+        tasks_to_run = intersect_tasks(available_tasks, requested_tasks)
+        test_configs.append(
+            (
                 "legacy_3.0.0_bundles",
                 V3_CASE_NAME,
                 expansions,
                 diff_dir_suffix,
                 tasks_to_run,
+            )
+        )
+
+    return test_configs
+
+
+def map_cfg_to_test_name(cfg: str) -> str:
+    """Map from weekly_* config names to actual test names."""
+    return cfg.replace("weekly_", "")
+
+
+def order_results(
+    test_results_dict: Dict[str, Results],
+    test_configs: List[tuple],
+    expansions: Dict,
+) -> Dict[str, Results]:
+    """Reorder results to match the order in expansions."""
+    ordered_results_dict: Dict[str, Results] = dict()
+
+    # Get the order from expansions
+    cfg_order = [map_cfg_to_test_name(cfg) for cfg in expansions["cfgs_to_run"]]
+    task_order = list(expansions["tasks_to_run"])
+
+    for cfg_name in cfg_order:
+        # Find the test config that matches this cfg_name
+        matching_config = None
+        for config in test_configs:
+            if config[0] == cfg_name:
+                matching_config = config
+                break
+
+        if matching_config is None:
+            continue
+
+        test_name = matching_config[0]
+        # Add results for each task in the order from expansions
+        for task in task_order:
+            # Look for exact key match: test_name_task
+            expected_key = f"{test_name}_{task}"
+            if expected_key in test_results_dict:
+                ordered_results_dict[expected_key] = test_results_dict[expected_key]
+
+    return ordered_results_dict
+
+
+def run_test(
+    test_name: str,
+    case_name: str,
+    expansions: Dict,
+    diff_dir_suffix: str,
+    tasks_to_run: List[str],
+) -> tuple[str, Dict[str, Results], str]:
+    """Run a single test and return its name, results dict, and captured output."""
+    captured_output = StringIO()
+
+    try:
+        test_results_dict: Dict[str, Results] = dict()
+
+        # Use context managers for thread-safe redirection
+        with redirect_stdout(captured_output), redirect_stderr(captured_output):
+            set_up_and_run_image_checker(
+                test_name,
+                case_name,
+                expansions,
+                diff_dir_suffix,
+                tasks_to_run,
                 test_results_dict,
             )
+
+        output = captured_output.getvalue()
+
+        # Write to individual log file
+        log_filename = f"test_{test_name}.log"
```

Comment:
Maybe make this `logs/...` so the logs are in a subdir for easy deletion (rather than `rm test_*.log`) or review.